src/results.rs

```diff
@@ -0,0 +1,33 @@
+use std::fmt;
+use std::error::Error;
+
+#[derive(Debug)]
+pub struct ParsingError {
+    desc: String,
```

Comment:
Not sure how I feel about "stringly-typed" errors for parsing. At the very least there are some standard parsing error states - Expected x and got y and Unexpected EOF for example - which could be modeled as an Enum.